tests/rptest/tests/nodes_decommissioning_test.py

```diff
@@ -422,6 +422,67 @@ def test_decommissioning_cancel_ongoing_movements(self):
 
         self.verify()
 
+    @cluster(num_nodes=6)
+    def test_learner_gap_metrics(self):
+        self.start_redpanda()
+        # set small segment size to calculate the gap correctly
+        self.redpanda.set_cluster_config({"log_segment_size": 1024 * 1024})
+        self._create_topics()
+
+        self.start_producer()
+        self.start_consumer()
+        # set recovery rate to small value to stop moves
+        self._set_recovery_rate(1)
+
+        def calculate_total_learners_gap() -> int | None:
+            gap = self.redpanda.metrics_sample("learners_gap_bytes")
+            if gap is None:
+                return None
+            return sum(g.value for g in gap.samples)
+
+        assert calculate_total_learners_gap(
+        ) == 0, "when there are no pending partition movements the reported gap should be equal to 0"
+
+        to_decommission = random.choice(self.redpanda.nodes)
+        to_decommission_id = self.redpanda.node_id(to_decommission)
+
+        self.logger.info(f"decommissioning node: {to_decommission_id}", )
+        self._decommission(to_decommission_id)
+
+        def learner_gap_reported(decommissioned_node_id: int):
+            total_gap = calculate_total_learners_gap()
+            p_size = self.redpanda.metrics_sample("partition_size")
+            if not total_gap or not p_size:
+                return False
+            total_size = sum([
+                ps.value for ps in p_size.samples
+                if self.redpanda.node_id(ps.node) == decommissioned_node_id
+            ])
```

Comment:
nit: square brackets still not needed

how it works:
1) `sum([...])` (sum of a list expression) is equivalent to `sum((...))` (sum of a generator expression), the latter just won't materialize the result
2) `sum((...))` is equivalent to `sum(...)` by a purely syntactic rule that double brackets can go single

Reply:
without square brackets `if` is not applied

Reply:
:scream: I'm fine merging it in any case, but quite curious why it may happen.
```
>>> sum(a**2 for a in range(10) if a % 2 != 0)
165
>>> sum(a**2 for a in range(10))
285
```